### PR TITLE
Broadcast last tick before leader rotation

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
When the leader is handing off the last tick before leader rotation, if that tick is dropped on the network, the next leader will not know to initiate his leader period.

#### Summary of Changes
Broadcast the last tick to the entire network before shutting down BroadcastStage for leader rotation.

Fixes #

### CODE REVIEW

File: src/broadcast_stage.rs [hunk 1]
```diff
@@ -52,6 +55,11 @@ fn broadcast(
     inc_new_counter_info!("broadcast_stage-entries_received", num_entries);
 
     let to_blobs_start = Instant::now();
+    *tick_height += ventries
+        .iter()
+        .flatten()
+        .fold(0, |tick_count, entry| tick_count + entry.is_tick() as u64);
```
Comment (garious) [inline src/broadcast_stage.rs]: `fold` is a sledgehammer. If a scalpel is available, use that. How about something like?

`flat_map(|entry| entry.is_tick() as u64).sum()`

File: src/cluster_info.rs [hunk 2]
```diff
@@ -502,7 +503,19 @@ impl ClusterInfo {
...
+            // despite packet drops)
+            if idx == received_index - 1 && contains_last_tick {
+                // If we see a tick at max_tick_height, then we know it must be the last
+                // Blob in the window, at index == received_index. There cannot be an entry
+                // that got sent after the last tick, guaranteed by the PohService).
+                let all: Vec<&NodeInfo> = broadcast_table.iter().collect();
+                assert!(window_l[w_idx].data.is_some());
+                orders.push((window_l[w_idx].data.clone(), all));
+            } else {
+                orders.push((window_l[w_idx].data.clone(), vec![&broadcast_table[br_idx]]));
```
Comment (garious) [inline src/cluster_info.rs]: Move this to after the if-else: `orders.push((window_l[w_idx].data.clone(), all));`?

File: src/cluster_info.rs [hunk 3]
```diff
@@ -522,42 +535,60 @@ impl ClusterInfo {
...
-                    blob.meta.size,
-                    v.id,
-                    v.contact_info.tvu
-                );
-                e
+                let send_errs_for_blob: Vec<_> = vs
+                    .iter()
+                    .map(move |v| {
+                        let e = s.send_to(&blob.data[..blob.meta.size], &v.contact_info.tvu);
+                        if log_enabled!(Level::Trace) {
```
Comment (garious) [inline src/cluster_info.rs]: A redundant check?

File: src/broadcast_stage.rs (hunk 1 above)
Comment (rob-solana) [reply src/broadcast_stage.rs]: ug, I hate that cast ;)

Comment (rob-solana) [reply src/broadcast_stage.rs]: because "entry.is_tick as u64" would also compile...

**Pull request merged**